apim-operator/pkg/controller/api/api_controller.go

```diff
@@ -260,6 +293,47 @@ func createMGWSecret(r *ReconcileAPI, confData string) error {
 
 }
 
+func getCredentials(r *ReconcileAPI, name string) error {
+
+	hasher := sha1.New()
+
+	//get the secret included credentials
+	credentialSecret := &corev1.Secret{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: name, Namespace: "wso2-system"}, credentialSecret)
+
+	if err != nil && errors.IsNotFound(err) {
+		fmt.Println("secret not found")
+		return err
+	}
+
+	//get the username and the password
+	for k, v := range credentialSecret.Data {
+		if strings.EqualFold(k, "username") {
+			basicUsername = string(v)
+			fmt.Println("basic username")
+			fmt.Println(basicUsername)
+		}
+		if strings.EqualFold(k, "password") {
+
+			//encode password to sha1
+			_, err := hasher.Write([]byte(v))
+			if err != nil {
+				return err
+			}
+
+			//convert encoded password to a hex string
+			basicPassword = hex.EncodeToString(hasher.Sum(nil))
+			fmt.Println("password hasher sum")
+			fmt.Println("basic password")
+			fmt.Println(basicPassword)
```

Comment:
do not print the password

Reply:
Is it fully encrypted?

Reply:
yeah. password is encrypted with sha1 since microgateway required sha1 encryption.